thread.go

```diff
@@ -0,0 +1,62 @@
+package pinbox
+
+import (
+	notmuch "github.com/msp301/go.notmuch"
+)
+
+type Thread struct {
+	ID         string    `json:"id"`
+	Type       string    `json:"type"`
+	Subject    string    `json:"subject"`
+	NewestDate int64     `json:"newestDate"`
+	OldestDate int64     `json:"oldestDate"`
+	Authors    []string  `json:"authors"`
+	Messages   []Message `json:"messages"`
+}
+
+func toOurThread(thr *notmuch.Thread) Thread {
```

Comment:
ooh ... actually `toOurThread()` and similar are actually a Notmuch specific implementation. This should probably live within `notmuch.go` in that case ... then hopefully that means all Notmuch specific stuff is in one place and theoretically if we ever wished to swap it out we should just need to implement a new `Mailbox` interface.

- [ ] Move toOurThread and similar into `notmuch.go`